ssl/quic/quic_channel.c

```diff
@@ -2559,13 +2575,30 @@ static void ch_raise_net_error(QUIC_CHANNEL *ch)
     ch_start_terminating(ch, &tcause, 1);
 }
 
+int ossl_quic_channel_net_error(QUIC_CHANNEL *ch)
+{
+    return ch->net_error;
+}
+
+void ossl_quic_channel_restore_err_state(QUIC_CHANNEL *ch)
+{
+    if (ch == NULL)
+        return;
+
+    OSSL_ERR_STATE_restore(ch->err_state);
+}
+
 void ossl_quic_channel_raise_protocol_error(QUIC_CHANNEL *ch,
                                             uint64_t error_code,
                                             uint64_t frame_type,
                                             const char *reason)
 {
     QUIC_TERMINATE_CAUSE tcause = {0};
 
+    if (error_code == QUIC_ERR_INTERNAL_ERROR)
+        /* Internal errors might leave some errors on the stack. */
```

Comment:
Should we push an error to the stack for any protocol error?

Reply:
Created #21337 for this decision.